cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -933,14 +933,20 @@ protected Future<Void> restart(Pod pod, RestartContext restartContext) {
     /* test */ KafkaAvailability availability(Admin ac) {
         return new KafkaAvailability(reconciliation, ac);
     }
-    
+
     /**
-     * Return true if the given {@code nodeId} is the active controller and there are other controllers we might yet have to consider.
+     * Checks if the given {@code nodeId} is the active controller and there are other controllers we might yet have to consider.
      * This ensures that the active controller is restarted/reconfigured last.
+     *
+     * @return true if the given {@code nodeId} is the active controller and there are other controllers we might yet have to consider.
      */
-    private boolean deferController(NodeRef nodeRef, RestartContext restartContext) throws Exception {
-        int controller = controller(nodeRef, operationTimeoutMs, TimeUnit.MILLISECONDS, restartContext);
-        if (controller == nodeRef.nodeId()) {
+    @SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE") // seems to be completely spurious
+    boolean deferController(NodeRef nodeRef, RestartContext restartContext) throws Exception {
+        int id = await(restartContext.quorumCheck.quorumLeaderId(), operationTimeoutMs, TimeUnit.MILLISECONDS,
+                t -> new UnforceableProblem("An error while trying to determine the quorum leader id", t));
```

Comment:
You replaced "quorum leader id" with "active controller" elsewhere. Maybe we should do that here as well?